### DISCUSSION TITLE
Patch Question

### DISCUSSION THREAD

**Discussion by ryanbuening** [ravendb/ravendb#17013] [Q&A] [answered]
I have the following Patch syntax

```sql
from Claims as c
where true and not exists(c.PolicyData.IsCommercial)
update {
   c.PolicyData.IsCommercial = false;
}
```

And in Raven Studio, I am running a Patch Test against a document ID in the Claims collection that has the following structure:

```json
{
    "PolicyData": {
        "IsCommercial": true
    },
    "@metadata": {
        "@collection": "Claims",
    }
}
```

The result of the Patch Test is that `IsCommercial` gets set to `false` on that document. I would expect that `IsCommercial` would **not** get updated as `PolicyData.IsCommercial` does exist on that document. What am I missing?

**Comment by ryanbuening** [comment]
It seems like the following works as I expect

```sql
from Claims as c
update {
    if (!c.PolicyData.IsCommercial) {
        c.PolicyData.IsCommercial = false;
    }
}
```

However, if I try to include a `where` clause to select certain documents (example below), the Patch will still update documents with IDs that do not start with `auto`

```sql
from Claims as c
where startsWith(id(), 'auto')
update {
	if (!c.PolicyData.IsCommercial) {
		c.PolicyData.IsCommercial = false;
	}
}
```

EDIT: The following seems to work. 🤷

```sql
from Claims as c
update {
	if (startsWith(id(this), "auto") && !c.PolicyData.IsCommercial) {
		c.PolicyData.IsCommercial = false;
	}
}
```

**Reply by ayende** [reply]
That should not be the case at all.
Can you create a test case? I just tried your starts with issue and I got:

![image](https://github.com/ravendb/ravendb/assets/116915/e76c6f69-2d1b-4443-9531-62fb45fdd563)

**Reply by ryanbuening** [reply]
I was using the "Patch Test" feature (entering document ID(s)) and was checking the Before / After. Are you seeing expected results using that? (RavenDB version 5.4.104)

![image](https://github.com/ravendb/ravendb/assets/812974/0a6fe40b-c308-4aac-9da0-515234ec33ec)

**Reply by garayx** [reply]
yes it works for me on latests RavenDB 5.4 stable
![image](https://github.com/ravendb/ravendb/assets/16625387/358ad4bb-ad4d-4ea9-a732-195e738f3ca7)

**Reply by ryanbuening** [reply]
So here is what I'm seeing when I add a `where` clause. It could be that I do not understand the logic of how `startsWith` is implemented.

![image](https://github.com/ravendb/ravendb/assets/812974/efcee2c7-8f8b-4fab-a1d1-ceee9d36d80c)

**Reply by ayende** [reply, marked as answer]
Okay, you are talking about two _separate_ issues.

You are testing the patch on a _specific_ document, we aren't applying any `where` clause here.

The `where` is used when executing the query, not testing the patch

**Reply by ryanbuening** [reply]
Ah I see. Thanks.
